### ISSUE TITLE
All the workers were blocked because of a single lock entry

### ISSUE THREAD

**Issue by severo** [huggingface/datasets-server#1587]
Unfortunately, I deleted the faulty lock entry, and I don't remember its value.

But every worker was trying to start the same job, but none could acquire the lock, thus, they all looped on the same job.

**Comment by severo** [comment]
Related? some jobs are not deleted after 40 minutes (zombies) as expected. They're kept as "started" forever. Currently:

```
> db.jobsBlue.find({status: "started", started_at: {"$lt": new Date(ISODate().getTime() - 1000*60*40)}}).count()
< 14
```

but they all have a `finished_at` value, meaning they have been finished at some time. They should not be in "started" status anymore.

```
> db.jobsBlue.find({status: "started", finished_at: null, started_at: {"$lt": new Date(ISODate().getTime() - 1000*60*40)}}).count()
< 0
```

It's issue #1467.

**Comment by severo** [comment]
Hopefully, it's fixed by #1637. Let's close, and we will reopen if the behavior occurs again.
